devito/passes/clusters/cse.py

```diff
@@ -90,12 +90,13 @@ def _cse(maybe_exprs, make, min_cost=1, mode='default'):
 
     while True:
         # Detect redundancies
-        counted = count(processed).items()
-        targets = OrderedDict([(k, estimate_cost(k, True)) for k, v in counted if v > 1])
+        counted = count(processed, None).items()
+        targets = OrderedDict([(k, estimate_cost(k[0], True))
+                               for k, v in counted if v > 1])
 
         # Rule out Dimension-independent data dependencies
         targets = OrderedDict([(k, v) for k, v in targets.items()
-                               if not k.free_symbols & exclude])
+                               if not k[0].free_symbols & exclude])
```

Comment:
Nitpick: we use `and` elsewhere in the codebase

Reply:
No that's not the same, `&` is the set intersection not the logical and that returns something different

Reply:
Fair, I hadn't spotted they were sets